File: frontend-web/src/data/cemacCountries.ts
export type CemacCountry = {
  code: string;
  nameKey: string;
};

/** Pays membres de la zone CEMAC (codes ISO 3166-1 alpha-2). */
export const CEMAC_COUNTRIES: CemacCountry[] = [
  { code: "cm", nameKey: "cameroon" },
  { code: "cf", nameKey: "car" },
  { code: "td", nameKey: "chad" },
  { code: "gq", nameKey: "equatorialGuinea" },
  { code: "ga", nameKey: "gabon" },
  { code: "cg", nameKey: "congo" },
];

export function cemacFlagUrl(countryCode: string, width = 40): string {
  return `https://flagcdn.com/w${width}/${countryCode.toLowerCase()}.png`;
}

File: mobile/lib/data/cemac_countries.dart
/// Pays CEMAC (codes ISO alpha-2).
class CemacCountry {
  const CemacCountry({required this.code, required this.nameFr, required this.nameEn});

  final String code;
  final String nameFr;
  final String nameEn;

  String label(bool isEnglish) => isEnglish ? nameEn : nameFr;
}

const cemacCountries = [
  CemacCountry(code: 'CM', nameFr: 'Cameroun', nameEn: 'Cameroon'),
  CemacCountry(code: 'CF', nameFr: 'République centrafricaine', nameEn: 'Central African Republic'),
  CemacCountry(code: 'TD', nameFr: 'Tchad', nameEn: 'Chad'),
  CemacCountry(code: 'GQ', nameFr: 'Guinée équatoriale', nameEn: 'Equatorial Guinea'),
  CemacCountry(code: 'GA', nameFr: 'Gabon', nameEn: 'Gabon'),
  CemacCountry(code: 'CG', nameFr: 'Congo', nameEn: 'Congo'),
];
